packages/driver/src/cypress/command_queue.ts

```diff
@@ -163,7 +198,21 @@ export class CommandQueue extends Queue<$Command> {
       let ret
       let enqueuedCmd
 
+      // Queries can invoke other queries - they are synchronous, and get added to the subject chain without
+      // issue. But they cannot contain commands, which are async.
+      // This callback watches to ensure users don't try and invoke any commands while inside a query.
```

Comment:
Not to block this, but interested to review the docs associated with queries (we will need to make sure how they work and requirements/limitations are clear to users, so they can author their own).

Reply:
After discussing this with Brian, we're holding off making the API public, which is why the PR switched to `_addQuery`, indicating an internal method. Documentation will wait until the end of the project, when we're ready to remove the underscore and have worked out all the kinks.

Reply:
Sorry I should clarify; I'll be interested in seeing it when it's available (not saying we should have it done now - definitely can come later).